### ISSUE TITLE
Windows test failures

### ISSUE THREAD

**Issue by jordansissel** [logstash-plugins/logstash-output-s3#7]
Originally from here: https://github.com/elasticsearch/logstash/issues/2487
```
  21) LogStash::Outputs::S3#register should create the tmp directory if it doesn't exist
     Failure/Error: FileUtils.rm_r(temporary_directory)
     Errno::EACCES:
       Permission denied - C:\Users\jls\AppData\Local\Temp/temporary_directory-a6f078be29a35da4c0affa10b36ee52394c8ea02121e58c52455ed46031f/ls.s3.sadness.2015-01-30T14.12.part0.txt
     # C:\Users\jls\Documents\GitHub\logstash\lib\logstash\runner.rb:57:in `run'
     # C:\Users\jls\Documents\GitHub\logstash\lib\logstash\runner.rb:112:in `run'
     # C:\Users\jls\Documents\GitHub\logstash\lib\logstash\runner.rb:170:in `run'

  22) LogStash::Outputs::S3#move_file_to_bucket should not upload the file if the size of the file is zero
     Failure/Error: s3.move_file_to_bucket(temp_file)
     ArgumentError:
       wrong number of arguments (0 for 1)
     # C:\Users\jls\Documents\GitHub\logstash\lib\logstash\runner.rb:57:in `run'
     # C:\Users\jls\Documents\GitHub\logstash\lib\logstash\runner.rb:112:in `run'
     # C:\Users\jls\Documents\GitHub\logstash\lib\logstash\runner.rb:170:in `run'

  23) LogStash::Outputs::S3#move_file_to_bucket should upload the file if the size > 0
     Failure/Error: s3.move_file_to_bucket(tmp)
     ArgumentError:
       wrong number of arguments (0 for 1)
     # C:\Users\jls\Documents\GitHub\logstash\lib\logstash\runner.rb:57:in `run'
     # C:\Users\jls\Documents\GitHub\logstash\lib\logstash\runner.rb:112:in `run'
     # C:\Users\jls\Documents\GitHub\logstash\lib\logstash\runner.rb:170:in `run'

  24) LogStash::Outputs::S3 when rotating the temporary file doesn't skip events if using the size_file option
     Failure/Error: pipeline = LogStash::Pipeline.new(config)
     LogStash::PluginLoadingError:
       Couldn't find any input plugin named 'generator'. Are you sure this is correct? Trying to load the generator input plugin resulted in this error: no such file to load -- logstash/inputs/generator
     # ./lib/logstash/plugin.rb:142:in `lookup'
     # ./lib/logstash/plugin.rb:140:in `lookup'
     # ./lib/logstash/pipeline.rb:270:in `plugin'
     # (eval):7:in `initialize'
     # ./lib/logstash/pipeline.rb:32:in `initialize'
     # C:\Users\jls\Documents\GitHub\logstash\lib\logstash\runner.rb:57:in `run'
     # C:\Users\jls\Documents\GitHub\logstash\lib\logstash\runner.rb:112:in `run'
     # C:\Users\jls\Documents\GitHub\logstash\lib\logstash\runner.rb:170:in `run'

  25) LogStash::Outputs::S3 when rotating the temporary file doesn't skip events if using the time_file option
     Failure/Error: Stud::Temporary.directory do |temporary_directory|
     Errno::EACCES:
       Permission denied - C:\Users\jls\AppData\Local\Temp/studtmp-7340c6a684273fb26a56eb96e962fdf1bdfbede5b978fe96dfb8f8fd4ceb/logstash-programmatic-access-test-object-1422655940
     # C:\Users\jls\Documents\GitHub\logstash\lib\logstash\runner.rb:57:in `run'
     # C:\Users\jls\Documents\GitHub\logstash\lib\logstash\runner.rb:112:in `run'
     # C:\Users\jls\Documents\GitHub\logstash\lib\logstash\runner.rb:170:in `run'

```

**Comment by ph** [comment]
Some file pointers are still open and cause this issue on windows.

**Comment by ph** [comment]
I've tracked the `LogStash::Outputs::S3 when rotating the temporary file doesn't skip events if using the time_file option` to the aws-sdk library, I am sending a string path to the `S3Object#write`. The library create an IO object to write chunk of the document to S3 and I think they are not closing the IO when using the `AWS.stub!`

**Comment by ph** [comment]
Same for `a6f078be29a35da4c0affa10b36ee52394c8ea02121e58c52455ed46031f/ls.s3.sadness.2015-01-30T14.12.part0.txt`

**Comment by ph** [comment]
The last failing test require the logstash-input-generator.

**Comment by ph** [comment]
windows fixes done in #9

**Comment by ph** [comment]
@jsvd Can you reviews this some windows test are failing because of this?
